src/olympia/activity/views.py

```diff
@@ -37,3 +47,41 @@ def check_object_permissions(self, request, obj):
         # Just loading the add-on object triggers permission checks, because
         # the implementation in AddonChildMixin calls AddonViewSet.get_object()
         self.get_addon_object()
+
+
+log = logging.getLogger('z.amo.mail')
+
+
+class NoAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        return request._request.user, None
+
+
+class EmailCreationPermission(object):
+    """Permit if client's IP address is allowed."""
+
+    def has_permission(self, request, view):
+        auth_token = request.META.get('HTTP_POSTFIX_AUTH_TOKEN', '')
+        if auth_token not in settings.POSTFIX_AUTH_TOKEN:
+            log.info('Invalid auth token [%s] provided' % (auth_token,))
+            return False
+
+        remote_ip = request.META.get('REMOTE_ADDR', '')
+        if remote_ip not in settings.ALLOWED_CLIENTS_EMAIL_API:
+            log.info('Request from invalid ip address [%s]' % (remote_ip,))
+            return False
+
+        return True
+
+
+@api_view(['POST'])
+@authentication_classes((NoAuthentication,))
```

Comment:
Probably don't need the `NoAuthentication` class, I haven't tested it but I suppose `@authentication_classes([])` would work.